src/components/checkbox/checkbox.js

```diff
@@ -171,12 +183,18 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       }
 
       function render() {
-        if(ngModelCtrl.$viewValue) {
+        if(ngModelCtrl.$viewValue && !isIndeterminate) {
           element.addClass(CHECKED_CSS);
         } else {
           element.removeClass(CHECKED_CSS);
         }
       }
+
+      function setIndeterminateState(newValue) {
+        isIndeterminate = (newValue === false) ? false : true;
+        element[0].indeterminate = isIndeterminate ? true : undefined;
```

Comment:
Apologies, I miss-understood the spec I was looking at. This is for when we move to the native checkbox. The only benefit of this would be that users can use the `:indeterminate` pseudo selector to find the indeterminate `md-checkbox`es.

@jelbourn I am okay with removing this code, do you care whether or not the directive is query-able using the  `:indeterminate` pseudo selector?